lib/rubygems.rb

```diff
@@ -1336,7 +1336,10 @@ def default_gem_load_paths
   # Defaults the operating system (or packager) wants to provide for RubyGems.
 
   require 'rubygems/defaults/operating_system'
-rescue LoadError
+rescue LoadError => e
+  raise e unless e.message.include? 'rubygems/defaults/operating_system'
```

Comment:
I think there's `LoadError#path` for this. Also, you can avoid generating a new exception and raise the original one with `raise` without arguments.